tests/components/plugwise/test_init.py

```diff
@@ -67,7 +69,42 @@ async def test_config_entry_not_ready(
     await hass.async_block_till_done()
 
     assert len(mock_smile_anna.connect.mock_calls) == 1
-    assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+    assert mock_config_entry.state == entry_state
+
+
+@pytest.mark.parametrize(
+    "side_effect",
+    [
+        (ConnectionFailedError),
+        (InvalidXMLError),
+        (ResponseError),
+        (UnsupportedDeviceError),
+    ],
+)
+async def test_async_update_fail_and_reconnect(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_smile_anna: MagicMock,
+    side_effect: Exception,
+) -> None:
+    """Test the Plugwise coordinator async_update failing."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    mock_smile_anna.async_update.side_effect = side_effect
+    coordinator = hass.data[DOMAIN][mock_config_entry.entry_id]
+    await coordinator.async_refresh()
+    await hass.async_block_till_done()
+
+    assert not coordinator.last_update_success
+    assert isinstance(coordinator.last_exception, UpdateFailed)
+
+    mock_smile_anna.async_update.side_effect = None
+    await coordinator.async_refresh()
+    await hass.async_block_till_done()
+
+    assert coordinator.last_update_success
```

Comment:
tbh, it is more interesting to test this on an entity level, to ensure the entities get the right state when an update has failed. I don't think testing the coordinator directly is a pattern we generally use.

Reply:
Can you point me to an integration that does this "correctly"?

Reply:
Fixed, @frenck thanks for the pointer on Discord :)